adapter/pkg/semanticversion/semantic_version.go

```diff
@@ -0,0 +1,109 @@
+/*
+ *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package semanticversion
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	logger "github.com/wso2/apk/adapter/pkg/loggers"
+)
+
+// SemVersion is the struct to store the version components of an API
+type SemVersion struct {
+	Version string
+	Major   int
+	Minor   int
+	Patch   *int
+}
+
+// ValidateAndGetVersionComponents validates version string and extracts version components
+func ValidateAndGetVersionComponents(version string, apiName string) (*SemVersion, error) {
```

Comment:
can't we do this validation in cr validation level (in validation webhook)?

Reply:
For Choreo there are APIs that don't match with this validation. Hence we have to support both API creations. That's why we didnt include this validation in the webhook.

Reply:
In there also we can check the config value and perform the validation right?